toonz/sources/toonz/levelcreatepopup.cpp

```diff
@@ -304,33 +304,42 @@ void LevelCreatePopup::updatePath() {
 void LevelCreatePopup::nextName() {
   const std::unique_ptr<NameBuilder> nameBuilder(NameBuilder::getBuilder(L""));
 
-  TLevelSet *levelSet =
-      TApp::instance()->getCurrentScene()->getScene()->getLevelSet();
-  ToonzScene *scene      = TApp::instance()->getCurrentScene()->getScene();
   std::wstring levelName = L"";
 
   // Select a different unique level name in case it already exists (either in
   // scene or on disk)
-  TFilePath fp;
-  TFilePath actualFp;
+
   for (;;) {
     levelName = nameBuilder->getNext();
 
-    if (levelSet->getLevel(levelName) != 0) continue;
-
-    fp       = scene->getDefaultLevelPath(getLevelType(), levelName);
-    actualFp = scene->decodeFilePath(fp);
-
-    if (TSystem::doesExistFileOrLevel(actualFp)) {
-      continue;
-    }
+    if (levelExists(levelName)) continue;
 
     break;
   }
 
   m_nameFld->setText(QString::fromStdWString(levelName));
 }
 
+//-----------------------------------------------------------------------------
+
+bool LevelCreatePopup::levelExists(std::wstring levelName) {
+  TFilePath fp;
+  TFilePath actualFp;
+  ToonzScene *scene = TApp::instance()->getCurrentScene()->getScene();
+  TLevelSet *levelSet =
+      TApp::instance()->getCurrentScene()->getScene()->getLevelSet();
+
+  fp       = scene->getDefaultLevelPath(getLevelType(), levelName);
```

Comment:
What do you think about changing this line to the following;
`TFilePath parentDir(m_pathFld->getPath().toStdWString());` 
`TFilePath fp = scene->getDefaultLevelPath(lType, levelName).withParentDir(parentDir);`
taking into account that levels can have the same name if they are saved in different folders?